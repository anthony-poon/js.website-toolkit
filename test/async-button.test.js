import React from "react";
import renderer from 'react-test-renderer';
import {AsyncButton} from "website-toolkit"

test("Render Test", () => {
    const component = renderer.create(
        <AsyncButton>Testing</AsyncButton>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
