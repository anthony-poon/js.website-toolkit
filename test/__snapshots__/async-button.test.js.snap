// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test 1`] = `
<button
  disabled={false}
  duration={1500}
  onClick={[Function]}
  type="button"
>
  Testing
</button>
`;

exports[`Render Test 2`] = `
<button
  disabled={true}
  duration={1500}
  onClick={[Function]}
  type="button"
>
  <span
    aria-hidden="true"
    className="spinner-border spinner-border-sm"
    role="status"
  />
</button>
`;
